subt/ros/proxy/artifact_type_from_string.cc

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2018 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <subt_ign/CommonTypes.hh>
+#include <subt_ign/protobuf/artifact.pb.h>
+
+
+// If we were willing to go all the way to C++17, I would consider returning
+// std::optional<subt::ArtifactType> instead of returning a bool and having
+// an output function parameter.
+bool artifactTypeFromString(const std::string& type_text,
+                            subt::ArtifactType& type_enum) {
+  // This may need c++11 or higher. I hope we support that, because anything
+  // below that is torturing ourselves without needing to.
+  //
+  // The static initialization also assumes single-threading. If multiple threads
+  // can call the conversion function at the same time, we can remove `static`
+  // and pay the cost of the function building the map at every call. Or do
+  // something else about it.
```

Comment:
This comment, however, is important. So delete "also" and keep it?

Reply:
fixed, thanks.